File: daily-ask/infrastructure/config/configure-message-brokers.go
package config

import (
	message_broker "github.com/sudo-von/daily-ask/infrastructure/message-brokers"
	"github.com/sudo-von/daily-ask/infrastructure/message-brokers/rabbitmq"
)

func ConfigureMessageBrokers(messageBrokerUrl string) {
	createQuestionProducer := rabbitmq.RabbitMQ{
		MessageBrokerUrl: messageBrokerUrl,
	}

	createQuestionProducer.Connect()

	createQuestionProducer.ProduceMessage(message_broker.CreateQuestionQueue, []byte("sasas"))
}

File: daily-ask/src/infrastructure/config/configure-message-brokers.ts
import LoggerService from '../services/logger-service/logger-service';
import RabbitMQCreateQuestionProducer from '../message-brokers/rabbitmq/producers/rabbitmq-create-question-producer';

const configureMessageBrokers = async (
  messageBrokerUrl: string,
  loggerService: LoggerService,
) => {
  const createQuestionProducer = new RabbitMQCreateQuestionProducer(
    messageBrokerUrl,
    loggerService,
  );

  await createQuestionProducer.connect();

  return { createQuestionProducer };
};

export default configureMessageBrokers;
